pkg/apis/elasticsearch/v1/elasticsearch_types.go

```diff
@@ -72,6 +72,22 @@ type ElasticsearchSpec struct {
 type TransportConfig struct {
 	// Service defines the template for the associated Kubernetes Service object.
 	Service commonv1.ServiceTemplate `json:"service,omitempty"`
+	// TLS defines options for configuring TLS on the transport layer.
+	TLS TransportTLSOptions `json:"tls,omitempty"`
+}
+
+type TransportTLSOptions struct {
+	// Certificate is a reference to a Kubernetes secret that contains the CA certificate
+	// and private key for generating node certificates.
+	// The referenced secret should contain the following:
+	//
+	// - `tls.crt`: The CA certificate.
+	// - `tls.key`: The private key for the CA certificate.
+	Certificate commonv1.SecretRef `json:"certificate,omitempty"`
```

Comment:
I was expecting a `name + namespace` combination here, with us copying the content over another secret in the ES namespace.
But we don't do that for the custom http cert, I don't think anyone complained.
So 👍 with only a name to keep things simple.